### ISSUE TITLE
Tasks should spawn on its own thread after execution

### ISSUE THREAD

**Issue by markhakansson** [rpa-tomorrow/substorm-nlp#142]
# Summary
After executing a task, the task should work on its own thread such that the application can still be used while the task is still active. 

# Examples
n/a

# Context
Currently if we set a reminder in 24 hours time, the program will be waiting for the reminder to complete before it's possible to do anything else in the application. Whether this should be handle by the interface itself or directly in the application would better be discussed.

**Comment by 97gushan** [comment]
When I try to reproduce the bug everything seems to work fine for me

**Comment by markhakansson** [comment]
Running the first reminder and then the second (not in the same command) then it works

**Comment by 97gushan** [comment]
The problem specified in this issue is fixed but because every task is prepared before the first one is executed then the last task overwrites the previous ones. Thus when "remind me in 5 seconds to go out. remind me in 10 seconds to eat." is inputted two reminders is run but both is executing at the same time (10s) with the same message (eat)

**Comment by markhakansson** [comment]
I agree. One way to solve it would be to have a completely separate executable class which the module will create after finishing the preparation of the task. Then the module can be reused without conflicting with the ready-to-be executed tasks. This increases complexity though and it might be difficult to make such a class abstract in such a way that it can be used for all prepared tasks across all modules.
